zypp/PoolQuery.cc

```diff
@@ -1387,6 +1402,9 @@ namespace zypp
           << it->idStr() << delim ;
     }
 
+    str << "comment: " << getComment() << delim ;
```

Comment:
I think if you'd use the PoolQueryAttr::commentAttr here instead of the string it would be better

Reply:
Also move the serializing of the comment to the end so that its the last element thats written to the file.

Reply:
> I think if you'd use the PoolQueryAttr::commentAttr here instead of the string it would be better

fix now

Reply:
> Also move the serializing of the comment to the end so that its the last element thats written to the stream.

I have made some update, correct me if any misunderstanding

Reply:
You fixed the first part, but still need to move the code down to the end of the function. Currently you write the comment in the middle of all the other properties, but since its a new property I'd prefer to write it as the last element to the stream,
before the `str << delim;` line

Reply:
Fixed.